cluster/node-pools/worker-karpenter/provisioners.yaml

```diff
@@ -155,25 +155,43 @@ spec:
             - "c7in"
             - "m7in"
             - "r7in"
+#{{ else if (gt (len .NodePool.InstanceTypes) 0) }}
+        - key: "node.kubernetes.io/instance-type"
+          operator: In
+          values:
+#      {{ range $type := .NodePool.InstanceTypes }}
+          - "{{ $type }}"
+#      {{ end }}
+#{{ end }}
+
+# safety guards to prevent the use of unwanted instance types in case the user has not specified any specific instance types
+#{{ if or (eq .NodePool.KarpenterInstanceTypeStrategy "default-for-karpenter") (eq .NodePool.KarpenterInstanceTypeStrategy "not-specified") }}
+        # exclude unwanted sizes
         - key: "karpenter.k8s.aws/instance-size"
           operator: "NotIn"
           values:
             - "metal"
-        # exclude instance-types with slow SSD
+        # exclude unwanted instance-types
         - key: "node.kubernetes.io/instance-type"
           operator: "NotIn"
           values:
-            - "c5d.large"
+            - "c5d.large" # too slow SSD
+        # user's custom configurations
```

Comment:
This looks like it wasn't merged/rebased quite right since: https://github.com/zalando-incubator/kubernetes-on-aws/pull/8406